src/apps/api/tests/test_competitions.py

```diff
@@ -0,0 +1,52 @@
+import json
+
+from django.urls import reverse
+from rest_framework.test import APITestCase
+
+from competitions.models import CompetitionParticipant
+from factories import UserFactory, CompetitionFactory, CompetitionParticipantFactory, PhaseFactory, LeaderboardFactory, \
+    ColumnFactory
+
+
+class CompetitionParticipantTests(APITestCase):
+    def setUp(self):
+        self.creator = UserFactory(username='creator', password='creator')
+        self.other_user = UserFactory(username='other_user', password='other')
+        self.comp = CompetitionFactory(created_by=self.creator)
+        PhaseFactory(competition=self.comp)
+        self.leaderboard = LeaderboardFactory(competition=self.comp)
+        ColumnFactory(leaderboard=self.leaderboard)
+
+    def _prepare_competition_data(self, url):
+        # data = CompetitionSerializer(comp).data
+        # data.pop('id')
+        # data.pop('logo')
+        # return data
+        resp = self.client.get(url)
+        data = resp.data
+        data.pop('id')
+        data.pop('logo')
+
+        # Just get the key from the task and pass that instead of the object
+        data["phases"][0]["tasks"] = [data["phases"][0]["tasks"][0]["key"]]
+        return data
+
+    # TODO: Do we have competition permissions tests?
+    # def test_cant_edit_someone_elses_competition?
+
+    def test_adding_organizer_accepts_them_as_participant(self):
+        CompetitionParticipantFactory(
```

Comment:
the user shouldn't need to already have a "pending" status for them to be added as a participant. should maybe test both use cases? that they get added if they haven't already applied and then accepted if they have.

Reply:
added a test for this other case as well ✔️